<?xml version="1.0"?>
<launch>
  <arg name="resolution" />
  <arg name="scatter_step_distance" />
  <arg name="scatter_steps" />
  <arg name="calibration_messages" />

  <group ns="kinect2">
    <node name="pose_estimator" pkg="tfpose_ros" type="broadcaster_ros.py" output="screen" required="true" >
      <param name="resolution" value="$(arg resolution)" />

      <env name="CUDA_VISIBLE_DEVICES" value="0" />
      <param name="camera" value="sd/image_color_rect" />
      <param name="model" value="mobilenet_thin" />
    </node>

    <node name="skeleton_to_3d" pkg="skeleton3d" type="skeleton_sensor" output="screen" required="true" >
      <param name="input_pose" type="str" value="pose_estimator/pose" />
      <param name="input_pointcloud" type="str" value="sd/points" />
      <param name="output_skeleton" type="str" value="/repository/skeleton_input" />
      <param name="camera_name" type="str" value="kinect2" />
      <param name="scatter_step_distance" type="int" value="$(arg scatter_step_distance)" />
      <param name="scatter_steps" type="int" value="$(arg scatter_steps)" />
      <param name="frame_id" type="str" value="kinect2_link" />
      <param name="number_of_calibration_messages" type="int" value="$(arg calibration_messages)" />
    </node>
  </group>
</launch>
